src/extension/infer.rs

```diff
@@ -664,7 +664,9 @@ impl UnificationContext {
                 let solution =
                     ExtensionSet::from_iter(self.get_constraints(&m).unwrap().iter().filter_map(
                         |c| match c {
-                            Constraint::Plus(x, other_m) if &m == other_m => Some(x.clone()),
+                            Constraint::Plus(x, other_m) if m == self.resolve(*other_m) => {
```

Comment:
I think the following from the PR description is worth adding as a comment here: " In this case, the variable (LHS) is updated to the merged value in UnificationContext::merge_equal_metas so we don't need to resolve that one."